### PR TITLE
cleaned and fixed up TokenList/Macro copy/move constructor/assignment

### PR DESCRIPTION
**Pull request by firewave**

### CODE REVIEW

File: simplecpp.cpp [hunk 1]
```diff
@@ -190,18 +190,18 @@ void simplecpp::Token::printOut() const
 simplecpp::TokenList::TokenList(std::vector<std::string> &filenames) : frontToken(NULL), backToken(NULL), files(filenames) {}
 
 simplecpp::TokenList::TokenList(std::istream &istr, std::vector<std::string> &filenames, const std::string &filename, OutputList *outputList)
-    : frontToken(NULL), backToken(NULL), files(filenames)
+    : TokenList(filenames)
```
Comment (danmar) [inline simplecpp.cpp]: delegated constructors is a C++11 feature.

Comment (amai2012) [reply simplecpp.cpp:193]: @firewave  Do you plan to update this PR?

Comment (firewave) [reply simplecpp.cpp:193]: sure... when I am done with the anniversary cake 🎂

Comment (firewave) [reply simplecpp.cpp]: @amai2012 finally...

File: simplecpp.cpp [hunk 2]
```diff
@@ -252,10 +253,11 @@ simplecpp::TokenList &simplecpp::TokenList::operator=(TokenList &&other)
 {
     if (this != &other) {
         clear();
-        backToken = other.backToken;
-        other.backToken = nullptr;
-        frontToken = other.frontToken;
-        other.frontToken = nullptr;
+		frontToken = other.frontToken;
```
Comment (danmar) [inline simplecpp.cpp:256]: tabs

**Pull request merged**